### ISSUE TITLE
`(int) <bool>` not detected as `int(0)|int(1)`

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#2548]
Typecasting a bool to an int should allow psalm to identify the value as `int(0)|int(1)`, since that's literally all it can be.

I could have a look into PRing this if you could give me a headsup as to where to go poking around in psalm?

```php
<?php
/**
* @param 0|1 $int
*/
function foo(int $int) : void {
    echo (string) $int;
}

foo((int) ((bool) ((int) '1')));
```
> Psalm output (using commit d5d4a18):
> 
> ERROR: ArgumentTypeCoercion - 9:5 - Argument 1 of foo expects int(0)|int(1), parent type int provided

https://psalm.dev/r/10d6fd4bbb (forgot to paste the link in when I copied the php then copied the error output, re: psalm/psalm.dev#7)

**Comment by SignpostMarv** [comment]
@psalm-github-bot go have a peek at psalm/psalm.dev#7 ¬_¬

**Comment by muglug** [comment]
> I could have a look into PRing this if you could give me a headsup as to where to go poking around in psalm?

It's here: https://github.com/vimeo/psalm/blob/0e39759ed63d747b9b78d7f12d68c66c96ba3c8d/src/Psalm/Internal/Analyzer/Statements/ExpressionAnalyzer.php#L553-L558

**Comment by muglug** [comment]
Now fixed
